irohad/consensus/yac/storage/impl/yac_common.cpp

```diff
@@ -48,6 +48,8 @@ namespace iroha {
         return nonstd::nullopt;
       }
 
+        //TODO: this code is somewhat fragile. How about just putting this logic in VoteMessage itself,
```

Comment:
This will imply that the hashing logic is set by the object, and not by algorithm (not data object), which does not seem to be the case.